packages/catalog-readmodel-writer/src/consumerServiceV2.ts

```diff
@@ -13,19 +13,29 @@ export async function handleMessageV2(
     .with({ type: "EServiceCloned" }, (msg) => msg.data.clonedEservice)
     .otherwise((msg) => msg.data.eservice);
 
-  await eservices.updateOne(
-    {
-      "data.id": message.stream_id,
-      "metadata.version": { $lt: message.version },
-    },
-    {
-      $set: {
-        data: eservice ? fromEServiceV2(eservice) : undefined,
-        metadata: {
-          version: message.version,
-        },
-      },
-    },
-    { upsert: true }
-  );
+  await match(message)
+    .with({ type: "EServiceDeleted" }, async (message) => {
+      await eservices.deleteOne({
+        "data.id": message.stream_id,
+        "metadata.version": { $lt: message.version },
+      });
+    })
+    .otherwise(
```

Comment:
this may lead to future problems because every new event would fall in the `otherwise` branch without warning the developer.
May we do an `exhaustive` pattern match on type?
I know that it would me more verbose but it is much safer

Reply:
Updated